test/python/tf_unittest_runner.py

```diff
@@ -39,89 +42,145 @@ def main():
     optional.add_argument(
         '--run_test',
         help=
-        "Runs the testcase and returns the output. Eg:math_ops_test.math_ops_test.DivNoNanTest.testBasic"
+        "Runs the testcase and returns the output. Eg:math_ops_test.DivNoNanTest.testBasic"
     )
     parser._action_groups.append(optional)
     arguments = parser.parse_args()
 
-    all_dirs_to_path(arguments.tensorflow_path)
-
     if (arguments.list_tests):
-        list_tests(arguments.list_tests)
-
+        test_list = helper(arguments.tensorflow_path, arguments.list_tests)
+        print('\n'.join(test_list))
     if (arguments.run_test):
-        run_test(arguments.run_test)
+        test_list = helper(arguments.tensorflow_path, arguments.run_test)
+        run_test(test_list)
+
+
+def helper(tf_path, test_regex):
```

Comment:
maybe a more descriptive name than `helper`. `get_atomic_test_list`?

Reply:
@sayantan-nervana I actually changed the name to get_test_list but did not go in previous commit. Updated now, please review.